Lib/test/test_importlib/import_/test_path.py

```diff
@@ -116,6 +116,31 @@ def test_None_on_sys_path(self):
             if email is not missing:
                 sys.modules['email'] = email
 
+    def test_bytes_on_sys_path(self):
+        # Putting bytes in sys.path[0] caused an import regression from Python
+        # 3.2: https://bugs.python.org/issue47025
+        with fixtures.tempdir() as tmp_path:
+            (tmp_path / "email.py").write_bytes(b"EXAMPLE = 'bytes_on_sys_path'\n")
+            new_path = sys.path[:]
+            new_path.insert(0, os.fsencode(tmp_path))
+            new_path_importer_cache = sys.path_importer_cache.copy()
+            new_path_hooks = [zipimport.zipimporter,
+                              self.machinery.FileFinder.path_hook(
+                                  *self.importlib._bootstrap_external._get_supported_file_loaders())]
+            missing = object()
+            email = sys.modules.pop('email', missing)
+            try:
+                with util.import_state(meta_path=sys.meta_path[:],
+                                       path=new_path,
+                                       path_importer_cache=new_path_importer_cache,
+                                       path_hooks=new_path_hooks):
+                    module = self.importlib.import_module('email')
```

Comment:
this results in a `BytesWarning: Comparison between bytes and string` when bytes are added to the `sys.path_importer_cache`

Reply:
Looks like that's causing CI to fail as `BytesWarning` is considered an error. Looks like I need to go to python-dev and ask what people want to do.

Reply:
https://discuss.python.org/t/drop-supporting-bytes-on-sys-path/17225

I'm now suggesting dropping bytes support in `sys.path`. We will see how that conversation goes.